packages/babel-traverse/src/hub.js

```diff
@@ -1,6 +1,14 @@
+import Scope from "./scope";
+export interface HubInterface {
+  mark?: (type: string, message: string) => void;
+  addHelper?: (name: string) => Object;
+  getScope?: () => Scope;
+  getCode?: () => string;
+  buildError:(node: Object, msg: string, Error: Error) => Error;
+}
+
 export default class Hub {
-  constructor(file, options) {
-    this.file = file;
-    this.options = options;
+  buildError(node, msg, BuildError = TypeError): Error {
+    return new BuildError(msg);
```

Comment:
@loganfsmyth what do you think about this approach? when `Error` is not specified, use `TypeError` as defaiult.
also changed its name to `BuildError` in the implementation, which is more descriptive IMO.